react/index.native.js

```diff
@@ -103,15 +103,25 @@ class Root extends Component<Props, State> {
     }
 
     /**
-     * Implements React's {@link Component#componentWillReceiveProps()}.
+     * Implements React's {@link Component#componentDidUpdate()}.
      *
      * New props can be set from the native side by setting the appProperties
      * property (on iOS) or calling setAppProperties (on Android).
      *
      * @inheritdoc
      */
-    componentWillReceiveProps({ url }) {
-        equals(this.props.url, url) || this.setState({ url: url || null });
+    componentDidUpdate(prevProps, prevState) {
+        // Ignore the special state update triggered on {@code Root}
+        // instantiation where an undefined url prop is set to a default.
+        if (typeof prevState.url === 'undefined'
+            && typeof this.state.url !== 'undefined') {
```

Comment:
Would it cause an issue if we still performed the state update here?

Reply:
I suppose not. Can't think of what would break here, actually.

Reply:
We need to thread lightly here, the idea as the current behavior is that navigating to the same URL we are currently on, it triggers a navigation.